src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -497,6 +498,91 @@ public static IEnumerable<object[]> GetAsync_IPBasedUri_Success_MemberData()
             Assert.True(connectionAccepted);
         }
 
+        [Theory]
+        [InlineData("321.123.456.789")]
+        [InlineData("321.123.456.789:8080")]
+        [InlineData("[::1234]")]
+        [InlineData("[::1234]:8080")]
+        public async Task ProxiedIPAddressRequest_NotDefaultPort_CorrectlyFormatted(string host)
+        {
+            string ipAddress = "http://" + host;
+            bool connectionAccepted = false;
+
+            await LoopbackServer.CreateClientAndServerAsync(async proxyUri =>
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    handler.Proxy = new WebProxy(proxyUri);
+                    try { await client.GetAsync(ipAddress); } catch { }
+                }
+            }, server => server.AcceptConnectionAsync(async connection =>
+            {
+                connectionAccepted = true;
+                List<string> headers = await connection.ReadRequestHeaderAndSendResponseAsync();
+                Assert.Contains($"GET {ipAddress}/ HTTP/1.1", headers);
```

Comment:
I wouldn't expect this test to ever work.   The "GET" line sent thru the proxy for a non-tunneled request would look more like this:

```
GET http://servername/ HTTP/1.1
Host: servername
```

etc.

So, you need to check for full URI with scheme such as:

>GET http://321.123.456.789:8080/ HTTP/1.1
Host: 321.123.456.789:8080

Reply:
See: https://tools.ietf.org/html/rfc7230#section-5.3.2

>When making a request to a proxy, other than a CONNECT or server-wide OPTIONS request (as detailed below), a client MUST send the target URI in absolute-form as the request-target.... An example absolute-form of request-line would be:
GET http://www.example.org/pub/WWW/TheProject.html HTTP/1.1

Reply:
It looks like the code adds the "http://" part above.  But yeah, it's confusing.

Reply:
Will rename the variable.